### DISCUSSION TITLE
Multiple Inputs for Single Output

### DISCUSSION THREAD

**Discussion by Djcjh** [tttapa/Control-Surface#383] [Q&A]
I wish to make a MIDI controller for my school final project and came across your code (thanks for creating it btw, it will really help out). I haven't taken a close look at it yet, but I would like to know if there is a way to have more than one input control a single output. For example, I wish to make a saxophone MIDI controller. With that I need to be able to hold down two or more buttons to play one note (holding down three buttons = middle G, holding down six buttons = middle D, etc.)

**Comment by tttapa** [comment]
The standard `NoteButton` class just reads a single input, independent of what the other buttons are doing.

You can certainly make something you describe, but you'll have to write some code yourself. You could use the [Send-MIDI-Notes](https://tttapa.github.io/Control-Surface-doc/Doxygen/d7/d8d/Send-MIDI-Notes_8ino-example.html) example as a starting point. All you need to do is add some more buttons and write some more `if` statements to detect when buttons are pressed simultaneously.

**Reply by Djcjh** [reply]
Thank you very much! I'll get right on it then; and I'll let you know if I have any more questions. 
P.S. thanks for replying so quickly!

**Reply by Djcjh** [reply]
I'm back with another quick question. I am wondering how to input sharp and flat notes. I see in the Send-MIDI-Notes that C, 4 was used. How would I go about adding C sharp? I'm not sure how the syntax is for it. Also, would I have to do all of my mapping in one of your classes, or can I do it all in my main code?

Thanks!

**Reply by tttapa** [reply]
You can use D flat (`note(Dd, octave)`) instead of C sharp. 

> Also, would I have to do all of my mapping in one of your classes, or can I do it all in my main code?

What do you mean? In 99% of cases, you shouldn't have to edit the library source files.

**Reply by Djcjh** [reply]
Ok, thanks! Don't worry, I haven't changed the libraries. I was just wanting to know how to set up my main code. I should be able to figure out the rest from here. Thanks again!
